compiler/cpp/src/generate/t_rs_generator.cc

```diff
@@ -290,13 +231,18 @@ void t_rs_generator::generate_typedef(t_typedef* ttypedef) {
   f_mod_ << "\n";
 }
 
+// Generates an enum, translating a thrift enum into a rust enum.
 void t_rs_generator::generate_enum(t_enum* tenum) {
   string ename = pascalcase(tenum->get_name());
-  indent(f_mod_) << "#[allow(dead_code)]\n";
-  indent(f_mod_) << "#[derive(PartialEq,Eq,Hash,Copy,Clone,Debug)]\n";
-  indent(f_mod_) << "pub enum " << ename << " {\n";
+  indent(f_mod_) << "enom! {\n";
   indent_up();
 
+  indent(f_mod_) << "name = " << ename << ",\n";
```

Comment:
I think it would be better style to define `endl` like the other generator implementations (i.e. `static const string endl = "\n";` and use that instead of `\n`. I don't think it should be too much work since you have much less code now and a few simple find and replaces or `sed` commands should do it, but we could also just make this change later.